File: docs/examples/2_register.py
import requests
import json

def register():
    url = "http://localhost:3000/auth/register"
    
    data = {
        "email": "newuser@example.com",
        "password": "SecurePass123!",
        "firstName": "Grace",
        "lastName": "Hopper",
        "role": "student"
    }
    
    try:
        response = requests.post(url, json=data)
        response.raise_for_status()
        print("Response:", response.json())
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")
        return None

# Usage
if __name__ == "__main__":
    register()

File: docs/examples/2_register.js
// Fetch API example (works in Node.js 18+ and browsers)
async function register() {
  try {
    const response = await fetch('http://localhost:3000/auth/register', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({"email":"newuser@example.com","password":"SecurePass123!","firstName":"Grace","lastName":"Hopper","role":"student"}),
    });

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();
    console.log('Response:', data);
    return data;
  } catch (error) {
    console.error('Error:', error);
  }
}

// Usage
register();